docs/narratives/build_backends.md

```diff
@@ -0,0 +1,228 @@
+`CFLAGS`/`LDFLAGS` and needs nothing else could look like:
+
+```bash
+# E.g., a default psabi-v3 usage with meson-python
+$ python -m build -Cvariant="psabi=x86-64-v3"
+```
+
+What actually happens under the hood for that build command? The build frontend,
+`build` in this case, only passes on the `-C` (shorthand for `--config-settings`)
+to the build backend. The build backend, `meson-python` in this case, may do the
+following:
+
+1. Query the plugins, find that `x86-plugin` provides the `psabi` variable, then
+   retrieve what that means in terms of `CFLAGS`/`CXXFLAGS`/`LDFLAGS` (that's
+   all one can probably support inside the plugin).
+2. Pass on those flags, either by amending those environment variables *or* by
+   translating them to CLI flags (e.g., C compiler flags can be passed like
+   `-Dc_args=xxx`, which is nicer for logging than amending environment
+   variables).
+3. Once the build is complete and the wheel is being assembled, modify the
+   wheel filename and modify the `METADATA` file to insert the variant content
+
+Now let's look at a more complicated example: building a NumPy wheel for
+`psabi=x86-64-v3`. The difference is that NumPy contains SIMD code and custom
+build flags to control whether and how that code gets included and used.
+The build backend cannot know this, hence the user invoking the build must
+instruct `meson-python` to only handle step (3) above, and not attempt to do
+(1) and (2). This is done by using the `-Cvariant-name` config setting, rather
+than `-Cvariant`:
+
+```bash
+# Build a NumPy psabi=x86-64-v3 wheel; this requires nonstandard flags (we want
+# to keep >v3 runtime-dispatched features)
+$ python -m build --wheel -Cvariant-name="psabi=x86-64-v3" -Csetup-args=-Dcpu-baseline=AVX2,FMA3 -Csetup-args=-Dcpu-dispatch=AVX512F,AVX512_SKX
+```
+
+Note that we've introduced two ways of specifying a variant build so far (which
+is probably all we'll need):
+
+- `-Cvariant=variable=value`: specify the wheel variant to build *and* let
+  the plugin add defaults (compile and link flags)
+- `-Cvariant-name=variable=value`: only specify the wheel variant to build
+
+So far so good for how to build x86-64 SIMD variants. Let's look at a second
+realistic case, namely BLAS variants. The difference is that we need not only
+compile and link flags, but also a different dependency - during the build,
+the correct shared library must be found *and* before the build, the user
+must have prepared the build environment correctly to ensure the shared library
+is actually present.
+
+Given the different dependencies here, and the relevant linker flags being
+different per build system, the plugin has no way to provide any compile/link
+flags here or do anything else that's helpful (it doesn't know after all which
+build backend we'll be using).
+
+Here is what the plugin's variables and values may be:
+
+```bash
+$ blas-plugin --list-variables
+blas
+lapack
```

Comment:
Do we ever want to use a different lapack provider than blas?

Reply:
Yes, for example BLIS is a BLAS-only library, so you'd use it with Netlib LAPACK or with libFlame. OpenBLAS can also be built as BLAS-only.